src/node_config.rs

```diff
@@ -191,6 +212,67 @@ pub fn cli_parse() -> CliConfig {
             .help("Disable LAN discovery"))
         .get_matches();
 
+    match matches.subcommand() {
+        ("config", Some(m)) => run_config(m),
+        _ => run_args(&matches),
+    }
+}
+
+/// Load settings from config file
+fn load_settings(settings: &mut Config) -> NodeConfig {
+    let udp_addr = settings.get::<SocketAddr>("udp-address").ok();
+    let tcp_addrs = settings.get::<Vec<SocketAddr>>("tcp-address").unwrap_or(Vec::new());
+    let keys_file = settings.get::<String>("keys-file").ok();
+    let bootstrap_nodes = settings.get::<Vec<BootstrapNode>>("bootstrap-nodes").unwrap_or(Vec::new());
+    let threads_config = settings.get::<ThreadsConfig>("threads").unwrap_or(ThreadsConfig::N(1));
+    let log_type = settings.get::<LogType>("log-type").unwrap_or(LogType::Stderr);
+    let motd = settings.get::<String>("motd").unwrap_or("This is tox-rs".to_owned());
```

Comment:
Can we use .merge to merge the default config with the config from file?

Reply:
https://github.com/tox-rs/tox-node/blob/da9c97c1015a1dfd149ed46715e9097199547b37/src/node_config.rs#L264

Reply:
1) new Config::default()
2) set config with default values.
3) read from config.yml then merge config
4) get settings from config then copy it to node_config.(get::<NodeConfig>())

Reply:
Approximately same complexity.

Reply:
Then there is no reason to set default values like "This is tox-rs" inside function called `load_settings`.

Reply:
There is `set_default` method that can help: https://docs.rs/config/0.9.1/config/struct.Config.html#method.set_default

Reply:
Or not :)
It seems it's useful only if we want to set defaults after we got a config object.